### ISSUE TITLE
Android force close when skuDetailList is no there due to invalid HTTP response on getItems()

### ISSUE THREAD

**Issue by tehong** [dooboolab/react-native-iap#76]
### Version of react-native-iap
0.2.13

### Expected behavior
getItems() should be able handle no skuDetailList

### Actual behavior
getItems() resulted in forced close when getItems() with no skuDetailList.

### Steps to reprodue the behabior
getItems() encountered HTTP error with no skuDetailList .

**It looks like we need more robust handling of the error when Android getItems() encounters HTTP error response.   I suggest that we return an reject with some kind of generic error message when skuDetailList is not there.**

**Looking through the latest master, it seemed that this error might be fixed.**

**If that's the case, you can close out this issue**

Here is the JAVA NullPointerException error.  Error is at RNIapModule.java 137 below:
```
java.lang.NullPointerException · Attempt to invoke interface method 'java.util.Iterator java.util.List.iterator()' on a null object reference
RNIapModule.java:137 com.reactlibrary.RNIapModule$3.onSkuDetailsResponse	
BillingClientImpl.java:404 com.android.billingclient.api.BillingClientImpl.querySkuDetailsAsync	
RNIapModule.java:128 com.reactlibrary.RNIapModule.getItems	
Method.java:-2 java.lang.reflect.Method.invoke	
JavaMethodWrapper.java:374 com.facebook.react.bridge.JavaMethodWrapper.invoke	
JavaModuleWrapper.java:162 com.facebook.react.bridge.JavaModuleWrapper.invoke	
NativeRunnable.java:-2 com.facebook.react.bridge.queue.NativeRunnable.run	
Handler.java:751 android.os.Handler.handleCallback	
Handler.java:95 android.os.Handler.dispatchMessage	
MessageQueueThreadHandler.java:31 com.facebook.react.bridge.queue.MessageQueueThreadHandler.dispatchMessage	
Looper.java:154 android.os.Looper.loop	
MessageQueueThreadImpl.java:194 com.facebook.react.bridge.queue.MessageQueueThreadImpl$3.run	
Thread.java:761 java.lang.Thread.run
```

**Comment by dooboolab** [comment]
@tehong Alright. So you are talking about NPE handling when there is no items. I'll support that. Also, it would be good to know that `0.3.0-alpha1` is released for more clean and reliable code base.

**Comment by tehong** [comment]
I've upgraded to 0.3.0-alpha1 I think that fixed my problem of HTTP response is not OK.  I'm not worry about no items at this point.  I am mainly interested in the HTTP response is not OK.

**Comment by dooboolab** [comment]
@tehong That is strange because we've not updated anything with the HTTP response since `0.2.*` to `0.3.*`. I think you should better check the connection to your play services or network. I am doubting you are living in `China` don't you?

**Comment by tehong** [comment]
@dooboolab No, I'm from US.  I'm talking about this line that was put in getItemBy* places on 03/05/2018:

```
+            if (responseCode == BillingClient.BillingResponse.OK) {
```

That should fix my problem with 0.2.13.  That line should guard against all getItemBy*() error response.

**Comment by dooboolab** [comment]
@tehong Thank you for sharing. That is already fixed in our newer version of `react-native-iap` which will be above `0.3.0-alpha1`. I've just published `0.3.1` also with some thorough testing. Just FYI.

**Comment by dooboolab** [comment]
I think this issue is closed for version above `0.3.0-alpha1` will close this issue.
